### PR TITLE
[pylint] Convert all report plugins to f-strings (part 1)

### PR DESCRIPTION
**Pull request by arif-ali**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [ ] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

File: sos/report/plugins/__init__.py [hunk 1]
```diff
@@ -134,32 +134,30 @@ def __str(self, quote=False, prefix="", suffix=""):
             optional prefix, suffix and value quoting.
         """
         quotes = '"%s"'
```
Comment (pmoravec) [inline sos/report/plugins/__init__.py:136]: Technically we will still use "s-string" due to this (but I think it does not matter?).

Comment (arif-ali) [reply sos/report/plugins/__init__.py:136]: Unless we make other wholesale changes.

At the moment I am looking to convert based on `pylint`, maybe we can have a another one later once parts 1, 2 and 3(in-coming) get merged

File: sos/report/plugins/__init__.py [hunk 2]
```diff
@@ -1763,24 +1752,22 @@ def get_filename_tag(fname):
...
                                 )
                                 continue
                         else:
                             self._log_debug(
-                                "stat of '%s' in container '%s' failed, "
-                                "skipping collection: %s"
-                                % (copyspec, con, ret['output'])
-                            )
+                                f"stat of '{copyspec}' in container '{con}' "
+                                f"failed,skipping collection: {ret['output']}")
```
Comment (pmoravec) [inline sos/report/plugins/__init__.py:1763]: Nitpick: missing space after comma.

File: sos/report/plugins/sapnw.py [hunk 3]
```diff
@@ -38,54 +38,50 @@ def collect_list_instances(self):
...
                 if not self.path_exists(path):
                     continue
                 for line in self.listdir(path):
                     if all(f in line for f in [sid, inst, vhost]):
-                        ldenv = 'LD_LIBRARY_PATH=/usr/sap/%s/SYS/exe/run' % sid
+                        ldenv = f'LD_LIBRARY_PATH=/usr/sap/{sid}/SYS/exe/run'
                         # Unicode is assumed here
                         # nuc should be accounted
-                        path = '/usr/sap/%s/SYS/exe/uc/linuxx86_64' % sid
+                        path = '/usr/sap/{sid}/SYS/exe/uc/linuxx86_64'
```
Comment (pmoravec) [inline sos/report/plugins/sapnw.py:49]: missing leading `f` (`path = f'/usr/sap/...`)

File: sos/report/plugins/block.py [hunk 4]
```diff
@@ -65,6 +65,6 @@ def setup(self):
             for line in lsblk['output'].splitlines():
                 if 'crypto_LUKS' in line:
                     dev = line.split()[0]
-                    self.add_cmd_output('cryptsetup luksDump /dev/%s' % dev)
+                    self.add_cmd_output(f'cryptsetup luksDump /dev/{dev}')
```
[the remark this answers is not in the document]
Comment (arif-ali) [reply sos/report/plugins/block.py:68]: yeah, there are many instances of that, while I have gone through all of these

File: sos/report/plugins/openstack_sahara.py [hunk 5]
```diff
@@ -52,13 +52,14 @@ def postproc(self):
         ]
         connection_keys = ["connection"]
 
+        join_con_keys = "|".join(connection_keys),
```
Comment (pmoravec) [inline sos/report/plugins/openstack_sahara.py:55]: Dangerous copy&paste typo, please remove the trailing comma (I know how it happened :), just let build the f-string regular expression for this code to see the problem..).

File: sos/report/plugins/openstack_trove.py [hunk 6]
```diff
@@ -51,13 +51,14 @@ def postproc(self):
         ]
         connection_keys = ["connection"]
 
+        join_con_keys = "|".join(connection_keys),
```
Comment (pmoravec) [inline sos/report/plugins/openstack_trove.py:54]: Ditto the comma.

File: sos/report/plugins/openstack_sahara.py (hunk 5 above)
Comment (arif-ali) [reply sos/report/plugins/openstack_sahara.py:55]: oops

File: sos/report/plugins/openstack_trove.py (hunk 6 above)
Comment (arif-ali) [reply sos/report/plugins/openstack_trove.py:54]: too much time staring at the screen to try and figure things out

**Review by pmoravec — APPROVED** [review]
Surprisingly low number of (already fixed) typos for such a big ant work!

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**